projects/com_protocol/src/main.c

```diff
@@ -0,0 +1,115 @@
+#include "can.h"
+#include "can_ack.h"
+#include "can_msg.h"
+#include "event_queue.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#define CAN_DEVICE_ID 5
+#define CAN_ID_A 0xA
+#define CAN_ID_B 0xB
+#define CAN_PERIOD_A_MS 1000
+#define CAN_PERIOD_B_MS 2000
+
+typedef enum {
+  TEST_CAN_EVENT_RX = 0,
+  TEST_CAN_EVENT_TX,
+  TEST_CAN_EVENT_FAULT,
+} TestCanEvent;
+
+static CanStorage s_can_storage = { 0 };
+
+CanMessage messageA = {
+  .source_id = CAN_DEVICE_ID,
+  .msg_id = CAN_ID_A,
+  .data = 4015,  // random uint16_t becomes uint64_t (not sure how to prevent)

```

Comment:
Here's the definition of `CanMessage` from `can_msg.h`:
```c
typedef struct CanMessage {
  uint16_t source_id;
  CanMessageId msg_id;
  union {
    uint64_t data;
    uint32_t data_u32[2];
    uint16_t data_u16[4];
    uint8_t data_u8[8];
  };
  CanMsgType type;
  size_t dlc;
} CanMessage;
```
So `data` is in a union with four other fields - fields in a union share the same storage, so you can think of them like different "views" into the blob of data they represent. Then to set a uint16_t, you'll have to set `data_u16[0]`.

Reply:
But isn't that just setting only the first variable in the array? Is it fine to leave the rest of the array as blank/null?

Reply:
Yes it's fine to leave the rest of the array as zeros, only the first `dlc` bytes should be TX'd.